src/components/Table/utils/TableState.ts

```diff
@@ -245,6 +281,32 @@ export class TableState {
   }
 
   setVisibleColumns(ids: string[]) {
+    // check to see if we have newly visible columns
+    if (ids.length > this.visibleColumnIds.length) {
+      // grab the new ids
+      const newlyVisibleIds = ids.filter((id) => !this.visibleColumnIds.includes(id));
+      // look through columns where we match ids to the new ids
+      const newExpanded = this.columns.filter((c) => newlyVisibleIds.includes(c.id) && c.initExpanded);
+      // here do a Promise.all() and execute column.expandColumns if necessary.
+      Promise.all(
+        newExpanded.map(async (c) => {
+          // if the column is initExpanded, manually toggle the column
+          if (c.initExpanded) {
+            this.toggleExpandedColumn(c.id);
+          }
```

Comment:
**thought:** Same thoughts as above... I'm kind of surprised we need to do this. Also, I believe every `column` that is in the `newExpanded` array would have `initExpanded: true` due to the filter before. So I think this conditional is unnecessary.